Formula/haskell-ide-engine.rb

```diff
@@ -0,0 +1,23 @@
+class HaskellIdeEngine < Formula
+  desc "The IDE engine and LSP server for Haskell. Not an IDE"
+  homepage "https://github.com/haskell/haskell-ide-engine"
+
+  url "https://github.com/haskell/haskell-ide-engine.git",
+      :revision => "8e0b60f3d47f6332c0f4415b9fff03815e229275"
+
+  version "0.1.0.0"
+  sha256 "e2195fc6bf01e4b9509e7be388e53fb4e0103658a3f26aca997b54cdd52273dd"
+
+  depends_on "haskell-stack" => :build
```

Comment:
Building with cabal is preferred and seems to be supported by upstream, is there a reason we can't use it?

Reply:
Downloading binary versions of GHC during the build is not permissible, so we'll probably need to pass on this formula.

Reply:
Is it be possible to provide a bottle/binary only version of this formula?

Reply:
It should be built from source with cabal using
```
depends_on "ghc"
```
if it's going to be in Homebrew.